### PR TITLE
Add support for AAD authentication using ADAL

### PR DESCRIPTION
**Pull request by ginach**
@daspek, @rgregg, @aclev, @mimisasouvanh

### CODE REVIEW

**Comment by mimisasouvanh** [comment]
@ginach It looks like this PR is going to revert the edits I made in collections.md, errors.md, and items.md, is that correct?

**Comment by ginach** [comment]
No, I wasn't trying to revert those changes. Must have happened with the merge. Will correct.

**Comment by ginach** [comment]
Should be fixed now!

**Comment by mimisasouvanh** [comment]
Great! Everything looks good to me.

**Comment by ginach** [comment]
Merging to dev, will add new PR for dev -> master

File: src/OneDriveSdk.Common/Authentication/DiscoveryService.cs [hunk 1]
```diff
@@ -1,14 +1,6 @@
-// ------------------------------------------------------------------------------
-//  **NOTE** This code was generated by a tool and will occasionally be
-//  overwritten. We welcome comments and issues regarding this code; they will be
-//  addressed in the generation tool. If you wish to submit pull requests, please
-//  do so for the templates in that tool.
-//
-//  This code was generated by Vipr (https://github.com/microsoft/vipr) using
-//  the T4TemplateWriter (https://github.com/msopentech/vipr-t4templatewriter).
-//
+﻿// ------------------------------------------------------------------------------
```
Comment (aclev) [inline src/OneDriveSdk.Common/Authentication/DiscoveryService.cs]: Why does it think that this file used to be DeltaResponse? Does DeltaResponse still exist?

Comment (ginach) [inline src/OneDriveSdk.Common/Authentication/DiscoveryService.cs]: I don't know. Git decided to treat a delete + add as a rename and the CL was enough files I didn't correct it. DeltaResponse does not exist, that metadata object gets transformed to a DeltaCollectionPage object in the templates.

**Pull request merged**